app/uk/gov/hmrc/cipphonenumberverification/connectors/CircuitBreakers.scala

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.cipphonenumberverification.connectors
+
+import akka.pattern.CircuitBreaker
+import akka.stream.Materializer
+import play.api.Logging
+import uk.gov.hmrc.cipphonenumberverification.config.CircuitBreakerConfig
+
+import scala.concurrent.Future
+import scala.concurrent.duration.Duration
+import scala.util.Try
+
+trait CircuitBreakers extends Logging {
+  def materializer: Materializer
+  def configCB: CircuitBreakerConfig
+
+  def withCircuitBreaker[T](block: => Future[T])(implicit connectionFailure: Try[T] => Boolean): Future[T] =
+    circuitBreaker.withCircuitBreaker(block, connectionFailure)
+
+  lazy val circuitBreaker = new CircuitBreaker(
+    scheduler = materializer.system.scheduler,
+    maxFailures = configCB.maxFailures, // Maximum number of failures before opening the circuit
+    callTimeout = configCB.callTimeout, // time after which to consider a call a failure
+    resetTimeout = configCB.resetTimeout, // time after which to attempt to close the circuit
+    maxResetTimeout = configCB.maxResetTimeout,
```

Comment:
add a comment for both maxResetTimeout and exponentialBackoffFactor to help understand this